.github/workflows/docker.yml

```
@@ -0,0 +1,35 @@
+name: Docker
+
+on:
+  pull_request:
+    branches: [ develop, main, release/** ]
+
+  # Allows you to run this workflow manually from the Actions tab
+  workflow_dispatch:
+
+jobs:
+  build:
+    name: Build and Run
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+        with:
+          submodules: recursive
+
+      - name: Update
+        run: |
+          sudo apt-get update
+
+      - name: Build the Docker compose
+        run: |
+          cd scripts
+          ./prepare.sh
+          docker compose run machine build
+
+      - name: Run the Docker compose
+        run: |
+          docker compose up -d
+          sleep 10
+          docker compose stop
```

```suggestion
          docker compose up -d wait
```

It's better to run `docker compose up -d --wait`rather than `up` and `stop`. It will exit with a non-zero exit code if not all containers will be running and healthy